redis_elasticache_client.go

```diff
@@ -23,10 +23,13 @@ type redisElastiCacheDB struct {
 }
 
 type config struct {
-	Username string `mapstructure:"username,omitempty"`
-	Password string `mapstructure:"password,omitempty"`
-	Url      string `mapstructure:"url,omitempty"`
-	Region   string `mapstructure:"region,omitempty"`
+	AccessKeyID     string `mapstructure:"access_key_id,omitempty"`
+	SecretAccessKey string `mapstructure:"secret_access_key,omitempty"`
+	SessionToken    string `mapstructure:"session_token,omitempty"`
+	Username        string `mapstructure:"username,omitempty"` // @Deprecated, use AccessKeyID instead
+	Password        string `mapstructure:"password,omitempty"` // @Deprecated, use SecretAccessKey instead
+	Url             string `mapstructure:"url,omitempty"`
+	Region          string `mapstructure:"region,omitempty"`
```

Comment:
nit: suggest to move the deprecated fields apart from the rest

```suggestion
	AccessKeyID     string `mapstructure:"access_key_id,omitempty"`
	SecretAccessKey string `mapstructure:"secret_access_key,omitempty"`
	SessionToken    string `mapstructure:"session_token,omitempty"`
	Url             string `mapstructure:"url,omitempty"`
	Region          string `mapstructure:"region,omitempty"`

	Username string `mapstructure:"username,omitempty"` // @Deprecated, use AccessKeyID instead
	Password string `mapstructure:"password,omitempty"` // @Deprecated, use SecretAccessKey instead
```